src/main/java/net/rptools/maptool/client/ui/htmlframe/HTMLOverlayPanel.java

```diff
@@ -217,25 +216,19 @@ private void removeOverlay(HTMLOverlayManager overlay) {
       AppMenuBar.removeFromOverlayMenu(overlay.getName());
       overlay.flush();
       if (overlays.isEmpty()) {
+        overlays.clear();
         setVisible(false); // hide overlay panel if all are gone
       }
     }
   }
 
   /** Removes all overlays. */
   public void removeAllOverlays() {
-    this.setVisible(false);
-    Platform.runLater(
-        () -> {
-          ObservableList<Node> listChildren = root.getChildren();
-          for (HTMLOverlayManager overlay : overlays) {
-            listChildren.remove(overlay.getWebView());
-            AppMenuBar.removeFromOverlayMenu(overlay.getName());
-            overlay.flush();
-          }
-          overlays.clear();
-          setVisible(false);
-        });
+    for (HTMLOverlayManager overlay : overlays) {
+      if (!overlay.getName().startsWith(AppConstants.INTERNAL_FRAME_PREFIX)) {
+        removeOverlay(overlay.getName());
+      }
+    }
```

Comment:
I think this needs to stay within `Platform.runLater()`, or at least the call to `removeOverlay()` needs to be.